packages/react-router/src/utils.ts

```diff
@@ -36,10 +38,12 @@ export type DeepPartial<T> = T extends object
     }
   : T
 
-// eslint-disable-next-line @typescript-eslint/naming-convention
-export type MakeDifferenceOptional<T, U> = Omit<U, keyof T> &
-  Partial<Pick<U, keyof T & keyof U>> &
-  PickRequired<Omit<U, keyof PickRequired<T>>>
```

Comment:
I'm not sure why this existed but it works without it?

Reply:
I added this here: 

https://github.com/TanStack/router/commit/92923cc4e08112f429b06f4b7120dc7241e8a2d8#diff-5fd7a0f05758a78cf8b73c09e4d71d21af63fe3d9b312b0ee75b04041ef078e7R222

do we have a test case that tests if when navigating e.g. from `/foo/$bar` to `/foo/$bar/baz/$id`, `$bar` is optional whereas `$id` is not?

Reply:
Yes. I thought this was covered by the code already here.

Reply:
This is the closest test case but maybe I should write a specific one!
https://github.com/TanStack/router/blob/10c717513d28ae2de918ff6f1cd74ff5734d6731/packages/react-router/tests/link.test-d.tsx#L253

Reply:
yeah would be cool to have that